src/requirements/Farcaster/components/FarcasterActionFollow.tsx

```diff
@@ -0,0 +1,37 @@
+import { RequirementButton } from "components/[guild]/Requirements/components/RequirementButton"
+import { useRequirementContext } from "components/[guild]/Requirements/components/RequirementContext"
+import useUser from "components/[guild]/hooks/useUser"
+import useFarcasterAction from "../hooks/useFarcasterAction"
+import { useFarcasterUser } from "../hooks/useFarcasterUsers"
+
+export default function FarcasterActionFollow() {
+  const { farcasterProfiles } = useUser()
+
+  const { data, roleId } = useRequirementContext<
+    "FARCASTER_FOLLOW" | "FARCASTER_FOLLOWED_BY"
+  >()
+
+  const { data: farcasterUser } = useFarcasterUser(data.id)
+
+  const { onSubmit, isLoading, response } = useFarcasterAction(roleId, "follow")
+
+  if (!farcasterProfiles || !farcasterProfiles?.[0]?.fid || response) {
```

Comment:
Do we actually need this first condition for some reason? Isn't the optional chaining enough for this check? 👀

Reply:
That's right, removed these checks in https://github.com/guildxyz/guild.xyz/pull/1322/commits/3f6cac54c1ca0ab3850fe48ec42bea2a0625a326